client/js/app.js

```diff
@@ -10,24 +10,65 @@ var Day = require('./components/day/day.js');
 var Week = require('./components/week/week.js');
 var Actionbar = require('./components/actionbar/action-bar.js');
 var Sidebar = require('./components/sidebar/sidebar.js');
+var classnames = require('classnames');
 
 var Locations = Router.Locations;
 var Location = Router.Location;
 var NotFound = Router.NotFound;
 
 var App = React.createClass({
+  getInitialState: function () {
+    return {
+      actionbarOpen: false,
+      hashRoute: ''
+    };
+  },
+
+  componentDidMount: function () {
+    window.addEventListener('hashchange', this.handleHashChange);
+  },
+
+  handleHashChange: function () {
+    var newHash = window.location.hash;
+    var actionbarHashActivated = /#actionbar?.+/.test(newHash);
```

Comment:
? is optionally matching the 'r', right? would /#actionbar.*/ more accurately describe what you are trying to accomplish? i think the one currently being used will match the string '/#actionbay' and other possibly unintended matches